test/files/pos/CustomCaseClassUnapply.scala

```diff
@@ -0,0 +1,19 @@
+package files.pos
+
+case class CaseClass(intData : Int, boolData : Boolean)
+object CaseClass{
+  def apply(intData: Int, boolData: Boolean): CaseClass = new CaseClass(intData, boolData)
+
+  def unapply(arg: CaseClass): scala.Option[Int] = Some(arg.intData)
+}
+
+object CustomCaseClassUnapply{
+
+  def main(args : Array[String]): Unit = {
```

Comment:
Because this currently compiles with unapply returning a tuple (the autotupling in pattern issue), I think a runtime test showing what actually happened is warranted. Maybe just wrap the block in println and `partest --update-check`. The pos test might be enough if you're confident about all the moving parts, which I am not because I just learned or maybe re-learned about them but only in part.